lib/nokogiri/xml/node.rb

```diff
@@ -192,16 +198,57 @@ def decorate!
 
       # :section: Manipulating Document Structure
 
-      ###
-      # Add +node_or_tags+ as a child of this Node.
-      #
-      # +node_or_tags+ can be a Nokogiri::XML::Node, a ::DocumentFragment, a ::NodeSet, or a String
-      # containing markup.
-      #
-      # Returns the reparented node (if +node_or_tags+ is a Node), or NodeSet (if +node_or_tags+ is
-      # a DocumentFragment, NodeSet, or String).
-      #
-      # Also see related method +<<+.
+      # :call-seq:
+      #   add_child(object) -> node or nodeset
+      #
+      # Appends child nodes to +self+.
+      #
+      # Argument +object+ is one of:
+      #
+      # - A Nokogiri::XML::Node (of any kind).
+      # - A Nokogiri::XML::NodeSet.
+      # - A string.
+      #
+      # When +object+ is a Nokogiri::XML::Node,
+      # adds the node as the last child of +self+;
+      # sets the parent of that node to +self+;
+      # returns the node:
+      #
+      #   doc = Nokogiri::XML::Document.parse('<root/>')
+      #   ele = Nokogiri::XML::Element.new('foo', doc)
+      #   doc.root.add_child(ele) # => #(Element: { name = "foo" })
+      #   doc.root.children.to_a  # => [#(Element: { name = "foo" })]
+      #   ele.parent == doc.root  # => true
+      #
+      # When +object+ is a Nokogiri::XML::NodeSet,
+      # appends each of its nodes to the children of +self+;
+      # returns the nodeset:
+      #
+      #   doc = Nokogiri::XML::Document.parse(BOOKSTORE_XML)
+      #   book_nodes = doc.search('//book')
+      #   other_doc = Nokogiri::XML::Document.parse('<other/>')
+      #   # Move all book nodes from doc to other_doc.root.
+      #   other_doc.root.add_child(book_nodes)
+      #   doc.search('//book').size       # => 0
+      #   other_doc.search('//book').size # => 4
+      #
+      # When +object+ is a string,
+      # creates a Nokogiri::XML::NodeSet object from the string;
+      # appends each of its nodes to the children of +self+;
+      # returns the nodeset:
+      #
+      #   doc = Nokogiri::XML::Document.parse('<root/>')
+      #   doc.root.add_child('<foo>FOO</foo>')
+      #   doc.root.children.to_a
+      #   # => [#(Element:0x593990 { name = "foo", children = [ #(Text "FOO")] })]
+      #   doc.root.add_child('<bar>BAR</bar><baz>BAZ</baz>')
+      #   doc.root.children.to_a
+      #   # =>
+      #   [#(Element:0x593990 { name = "foo", children = [ #(Text "FOO")] }),
+      #    #(Element:0x5992dc { name = "bar", children = [ #(Text "BAR")] }),
+      #    #(Element:0x5994bc { name = "baz", children = [ #(Text "BAZ")] })]
+      #
+      # Raises ArgumentError if +object+ is not a node, a nodeset, or a string.
```

Comment:
In general, I think information about return types, exceptions, and blocks should be the top of the docstring near the information about argument types.